### PR TITLE
Bridge deposit transition

### LINKED ISSUE #17: Implement deposit lock script
# Bounty

## Scope

- We need a lock script for deposits
- It unlocks either with a signature of the depositor, or if it is being spent by bridge typescript with id

## Deliverables

- The contract
- Some unit tests

## Funding Circle

Nervos

## Bounty Owner/Gardener

@TheReturnOfJan  as CKB Dev Lead
## Roles
bounty gardener: @TheReturnOfJan  / 50 DAI 
bounty worker: @TheReturnOfJan  / 80%
bounty reviewer: name / share

## Gardener checklist

- [x] I have attached one of the `size-` labels to this bounty
- [ ] I have submitted this bounty via http://bounty.leapdao.org/viewform
- [ ] I have added this bounty to bounty board here https://github.com/orgs/leapdao/projects/6

### PR DESCRIPTION
**Pull request by TheReturnOfJan**
Resolves #17 and #46.

### CODE REVIEW

File: packages/parent-bridge/contracts/parent-bridge/src/main.rs [hunk 1]
```diff
@@ -47,6 +47,11 @@ const ANYONE_CAN_PAY_CODE_HASH: [u8; 32] = [
     230, 131, 176, 65, 57, 52, 71, 104, 52, 132, 153, 194, 62, 177, 50, 109, 90, 82, 214, 219, 0,
     108, 13, 47, 236, 224, 10, 131, 31, 54, 96, 215,
 ];
+const DEPOSIT_LOCK_CODE_HASH: [u8; 32] = [
```
Comment (mwaa) [inline packages/parent-bridge/contracts/parent-bridge/src/main.rs:50]: Is this the value of code hash in test nest or live? Also hard coding this prevents local development is there a way to reuse or read from .env file

File: packages/parent-bridge/contracts/parent-bridge/src/main.rs [hunk 2]
```diff
@@ -104,17 +121,36 @@ impl StateTransition {
...
+                    .zip(QueryIter::new(load_cell_capacity, Source::Input))
+                    .filter(|(script, _)| {
+                        *script.code_hash().raw_data() == DEPOSIT_LOCK_CODE_HASH[..]
+                    })
+                    .map(|(_, cap)| cap)
+                    .sum();
+                let data_before = load_cell_data(0, Source::Input)?;
+                let data_after = load_cell_data(0, Source::Output)?;
+                Ok(Self::CollectDeposits {
+                    total: total_deposit_capacity,
```
Comment (mwaa) [inline packages/parent-bridge/contracts/parent-bridge/src/main.rs:147]: Informational question? Are these values attributed to the parent bridge cell? Is there a way to read them from cell data or we can just read the capacity of bridge cell only?

**Review by mwaa — CHANGES_REQUESTED** [review]

**Comment by mwaa** [comment]
Some tests are failing if you could fix those

capsule test failures:
-  tests::test_collect_1_deposit
-  tests::test_collect_3_deposits
-  tests::test_collect_deposit_fiddling_with_data

File: packages/parent-bridge/contracts/parent-bridge/src/main.rs (hunk 1 above)
Comment (TheReturnOfJan) [reply packages/parent-bridge/contracts/parent-bridge/src/main.rs:50]: No.

File: packages/parent-bridge/contracts/parent-bridge/src/main.rs (hunk 2 above)
Comment (TheReturnOfJan) [reply packages/parent-bridge/contracts/parent-bridge/src/main.rs:147]: Not sure I understand the question.

**Comment by TheReturnOfJan** [comment]
All the tests pass for me. Have you run capsule build?

**Review by mwaa — APPROVED** [review]
Great work on the rest.
Forked new issue to address running tests on different machines 
https://github.com/leapdao/nervos/issues/50

**Pull request merged**